docs/contributing/review-process.md

```diff
@@ -0,0 +1,172 @@
+review:
+
+1. Make sure your PR is no longer marked as a [draft][github-help-draft-pr], and
+   doesn't have "WIP" in the title.
+
+1. Post a quick "Ready for review!" comment on the main GitHub thread for your
+   PR.
+
+[github-help-draft-pr]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests
+
+## Stages of a pull request review
+
+This section describes the stages of the pull request review process. Each stage
+may require several rounds of iteration. Don't feel daunted! Not every PR will
+require all the stages described, and the process often goes quite quickly for
+smaller changes that are clearly explained.
+
+1. **Product review.** Oftentimes, seeing an initial implementation will make it
+   clear that the product design for a feature needs to be revised, or that
+   additional changes are needed. The reviewer may therefore ask you to amend or
+   change the implementation. Some changes may be blockers for getting the PR
+   merged, while others may be improvements that can happen afterwards. Feel
+   free to ask if it's unclear which type of feedback you're getting.
+   (Follow-ups can be a great next issue to work on!)
+
+   Your PR might be assigned the [product
+   review](https://github.com/zulip/zulip/pulls?q=is%3Aopen+is%3Apr+label%3A%22product+review%22)
+   label in this phase, or later in the review process as questions come up. You
+   can also add this label yourself if appropriate. If doing so, be sure to
+   clearly outline the product questions that need to be addressed.
+
+2. **QA.** If your PR makes user-facing changes, it may get a round of testing
+   without reference to the code. You will get feedback on any user-facing bugs
+   in the implementation. To minimize the number of review round-trips, make
+   sure to [thoroughly test](../contributing/code-reviewing.md#manual-testing)
+   your own PR prior to asking for review.
+
+   Your PR might be assigned the [QA
+   needed](https://github.com/zulip/zulip/pulls?q=is%3Aopen+is%3Apr+label%3A%22QA+needed%22)
+   label in this phase, or later on if re-testing is required.
+
+3. **Initial code review.** All PRs will go through one or more rounds of code
+   review. Your code may initially be [reviewed by other
+   contributors](../contributing/code-reviewing.md). This helps us make good use
+   of project maintainers' time, and helps you make progress on the PR by
+   getting quick feedback. A project maintainer may leave a comment asking
+   someone with expertise in the area you're working on to review your work.
+
+4. **Maintainer code review.** In this phase, a Zulip maintainer will do a
+   thorough review of your proposed code changes. Your PR may be assigned the
+   [maintainer
+   review](https://github.com/zulip/zulip/pulls?q=is%3Aopen+is%3Apr+label%3A%22maintainer+review%22)
+   label at this stage.
```

Comment:
Would it not be assigned this label in the case it was reviewed and thought good to go for the next labels?